src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -0,0 +1,150 @@
+# coding=utf-8
+# Copyright 2021 Facebook AI Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" MaskFormer model configuration"""
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Tuple
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+MASKFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    # "facebook/detr-resnet-50": "https://huggingface.co/facebook/detr-resnet-50/resolve/main/config.json",
+    # See all MaskFormer models at https://huggingface.co/models?filter=detr
+}
+
+logger = logging.get_logger(__name__)
+
+RGB = Tuple[int, int, int]
+
+
+@dataclass
+class ClassSpec:
+    is_thing: bool
+    label: str
+    colour: RGB
+
+
+@dataclass
+class DatasetMetadata:
+    classes: List[ClassSpec] = field(default_factory=list)
+
+
+class MaskFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MaskFormer`]. It is used to instantiate a MaskFormer
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the "maskformer_swin_base_IN21k_384_bs16_160k_res640" architecture trained on ade20k-150
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+    """
+    model_type = "mask_former"
+
+    attribute_map = {"hidden_size": "d_model"}
+
+    def __init__(
+        self,
+        dataset_metadata: DatasetMetadata = None,
+        fpn_feature_size: Optional[int] = 256,
+        mask_feature_size: Optional[int] = 256,
+        num_classes: Optional[int] = 150,
+        no_object_weight: Optional[float] = 0.1,
+        num_queries: Optional[int] = 100,
+        swin_pretrain_img_size: Optional[int] = 384,
+        swin_in_channels: Optional[int] = 3,
```

Comment:
```suggestion
        num_channels: Optional[int] = 3,
```
All existing vision models currently use num_channels regarding number of input channels.

Reply:
I'll change it as soon as I bring our Swin Transformer in the implementation